src/test/java/com/iota/iri/controllers/TransactionViewModelTest.java

```diff
@@ -481,4 +482,36 @@ public void updatedTxShouldBeSavedToDbBeforeTangleShutdown() throws Exception {
             Assert.assertTrue("TVM should be a milestone", tvm.isMilestone());
         }
     }
+
+    @Test
+    public void deletedTxFromDbShouldDeleteFromCache() throws Exception {
+        Cache<Indexable, TransactionViewModel> cache = tangle.getCache(TransactionViewModel.class);
+
+        Transaction tx = new Transaction();
+        Hash hash = HashFactory.TRANSACTION
+                .create("TEST9TRANSACTION9TEST9TRANSACTION9TEST9TRANSACTION9TEST9TRANSACTION9TEST999999999");
+        tx.address = hash;
+        TransactionViewModel tvm = new TransactionViewModel(tx, hash);
+
+        tvm.store(tangle, snapshot);
+        tvm.delete(tangle);
+        Assert.assertNull("TVM should be null", cache.get(hash));
```

Comment:
i think that if you move this line to the other test, and delete this test its also sufficient
no need to do an almost equal test twice.

Unless you see a case where `deletedTxFromDbShouldDeleteFromCache` fails but `deletedTxShouldNotBeBroughtBackToCache` succeeds?

Reply:
I am still waiting for @acha-bill's response.
But imo it makes sense that there are two tests, because the second test came to ensure that a bug we had won't come back...

And there is a case where one test fails and the other won't. If the second fail test and the first one won't we will know immediately that the problem was `TransactionViewModel.fromHash` is to blame

However, Brord has a point that since we unfortunately are not mocking `tangle` atm this test is expensive. The situation where we have poorly designed unit tests is less of a priority because they still work fine...

So I'll let @achabill make the final call

Reply:
I get what you mean @kwek20 .
But 2 tests makes it explicit. Like Gal said, there was an issue we had were deleted tx coming back to cache when `fromHash` was called.
So the 2 tests look fine for me.